### PR TITLE
Non-technical description of deep learning

### LINKED ISSUE #421: "Defining deep learning"
I think that this section doesn't quite accomplish what it should, in terms of highlighting deep learning's distinctive features and its relationship to "shallow" approaches that readers might be familiar with.  I have a small outline below.

I've tried rewriting it a few times, but I'm feeling stuck.  If someone agrees that these would be improvements and wants to try them out, go for it.  Otherwise I'll try again in a few days.

* Should mention the key features of deep learning, compared to other methods: 
    * composition simple functions in "layers"
    * differentiability/backpropagation (in most cases)
    * automated feature engineering
* Should clarify that deep learning does many of the same jobs as more familiar approaches, such as clustering, regression/classification. It just has the potential to do them better because (with sufficiently large data sets) these models can learn which features to use.
* This is very minor, but I think that the current wording implies that autoencoders are "new"

### PR DESCRIPTION
**Pull request by cgreene**
This is an attempt to solve the "what actually qualifies as deep learning" question. This would close #421. @davharris please fell free to edit this. I'd love to improve what's here.

### CODE REVIEW

File: sections/02_intro.md [hunk 1]
```diff
@@ -23,15 +23,25 @@ algorithms have produced comparable or higher accuracy than previous
...
-These methods are powered by improvements to artificial neural networks, an idea
-that was first proposed in 1943 [@doi:10.1007/BF02478259] as a model for how our
-brains process information. Since then, interest in neural networks as
-computational models has waxed and waned over time. This history is interesting
-in its own right [@doi:10.1103/RevModPhys.34.135]. In recent years, attention
-has shifted back to neural networks as hardware improvements have allowed the
-techniques that have come to be known as deep learning to surpass other machine
-learning algorithms. `TODO: add what is/isn't deep learning to this paragraph?
-Revise paragraph to also fix lengthy closing sentence. Issue #421`
+Deep learning methods have some common features. First, they encode data via
```
Comment (agapow) [inline sections/02_intro.md]: "layer" is introduced with no explanation

Comment (agitter) [inline sections/02_intro.md]: Part of the reason this is hard to write is that we're doing it without any figures.  I agree that layer isn't fully explained, but I struggle to explain it concisely with text alone.  We could consider leaving this and work on adding an introductory figure here during revision.

File: sections/02_intro.md [hunk 2]
```diff
@@ -23,15 +23,25 @@ algorithms have produced comparable or higher accuracy than previous
...
+suitable for many tasks and to customize those features for one or more specific
+tasks. Deep learning does many of the same things as more familiar approaches.
+Like a clustering algorithm it can build features that describe recurrent
+patterns in data. Like a regression approach deep learning methods can predict
+some output. However, deep learning methods combine both of these steps. When
+sufficient data are available, these methods construct features tuned to a
+specific problem and combine those features into a predictor. These methods are
+powered by improvements to artificial neural networks, an idea that was first
+proposed in 1943 [@doi:10.1007/BF02478259] as a model for how our brains process
+information. Since then, interest in neural networks as computational models has
```
Comment (agitter) [inline sections/02_intro.md]: Now that the paragraph is longer, could condense this

> Since then, interest in neural networks as computational models has waxed and waned over time [@doi:10.1103/RevModPhys.34.135].

File: sections/02_intro.md [hunk 3]
```diff
@@ -23,15 +23,25 @@ algorithms have produced comparable or higher accuracy than previous
...
-that was first proposed in 1943 [@doi:10.1007/BF02478259] as a model for how our
-brains process information. Since then, interest in neural networks as
-computational models has waxed and waned over time. This history is interesting
-in its own right [@doi:10.1103/RevModPhys.34.135]. In recent years, attention
-has shifted back to neural networks as hardware improvements have allowed the
-techniques that have come to be known as deep learning to surpass other machine
-learning algorithms. `TODO: add what is/isn't deep learning to this paragraph?
-Revise paragraph to also fix lengthy closing sentence. Issue #421`
+Deep learning methods have some common features. First, they encode data via
+multiple layers. Each layer essentially performs feature construction for the
```
Comment (agitter) [inline sections/02_intro.md]: "for the next layers"?  I've seen networks drawn top-to-bottom or bottom-to-top, which makes "below" ambiguous.

File: sections/02_intro.md [hunk 4]
```diff
@@ -23,15 +23,25 @@ algorithms have produced comparable or higher accuracy than previous
...
+layers below it. A process is often used to allow layers deeper in the network
```
Comment (agitter) [inline sections/02_intro.md]: "A training process" or "A parameter-estimation process"?

File: sections/02_intro.md [hunk 5]
```diff
@@ -23,15 +23,25 @@ algorithms have produced comparable or higher accuracy than previous
...
+to contribute to the refinement of earlier layers. This elements provide these
+algorithms with the capability to automatically engineer features that are
+suitable for many tasks and to customize those features for one or more specific
+tasks. Deep learning does many of the same things as more familiar approaches.
+Like a clustering algorithm it can build features that describe recurrent
```
Comment (agitter) [inline sections/02_intro.md]: I suggest a comma after algorithm

File: sections/02_intro.md [hunk 6]
```diff
@@ -23,15 +23,25 @@ algorithms have produced comparable or higher accuracy than previous
...
+patterns in data. Like a regression approach deep learning methods can predict
```
Comment (agitter) [inline sections/02_intro.md]: Comma after approach

File: sections/02_intro.md (hunk 1 above)
Comment (davharris) [inline sections/02_intro.md]: Layers are like steps in a pipeline, which most readers will be familiar with. The key feature of deep learning is that the whole pipeline can be optimized jointly by gradient descent (usually by backpropagation).

Comment (cgreene) [inline sections/02_intro.md]: I switched to "before" framing instead of "below"

File: sections/02_intro.md (hunk 4 above)
Comment (cgreene) [inline sections/02_intro.md]: 👍

File: sections/02_intro.md (hunk 5 above)
Comment (cgreene) [inline sections/02_intro.md]: 👍

File: sections/02_intro.md (hunk 6 above)
Comment (cgreene) [inline sections/02_intro.md]: 👍

File: sections/02_intro.md (hunk 2 above)
Comment (cgreene) [inline sections/02_intro.md]: 👍

**Comment by cgreene** [comment]
ok, pushed some attempt to address these concerns. i'll be on a flight & traveling up to a retreat today so i may be a bit in and out of contact. feel free to merge & file a second PR if that's easiser, or i can address later this afternoon.

**Comment by agitter** [comment]
This looks good.  I think the a few sentences starting with "These methods are powered" were intended to be cut.  I'll cut them and merge.
